src/storage/src/store.rs

```diff
@@ -311,6 +311,8 @@ pub trait LocalStateStore: StaticSendSync {
         key_range: IterKeyRange,
         read_options: ReadOptions,
     ) -> Self::MayExistFuture<'_>;
+
+    fn on_vnode_stale(&mut self);
```

Comment:
Seems that in this PR, we will let local state store be aware of whether it is singleton and get notified on vnode update. If so, I think we can directly let the local state store to be aware of the vnode mapping, since the vnode information will be useful in future development.

Then we may pass the vnode bitmap in local state store initialization, and replace the current `on_vnode_stale` with `update_vnode_bitmap` to let the local state store to keep track of the vnode mapping of itself all the way.

Reply:
> Seems that in this PR, we will let local state store be aware of whether it is singleton and get notified on vnode update. If so, I think we can directly let the local state store to be aware of the vnode mapping, since the vnode information will be useful in future development.
> 
> Then we may pass the vnode bitmap in local state store initialization, and replace the current `on_vnode_stale` with `update_vnode_bitmap` to let the local state store to keep track of the vnode mapping of itself all the way.

Unfortuanately we cannot determine whether a table is singleton from vnode mapping 😢

Reply:
> Seems that in this PR, we will let local state store be aware of whether it is singleton and get notified on vnode update. If so, I think we can directly let the local state store to be aware of the vnode mapping, since the vnode information will be useful in future development.
> 
> Then we may pass the vnode bitmap in local state store initialization, and replace the current `on_vnode_stale` with `update_vnode_bitmap` to let the local state store to keep track of the vnode mapping of itself all the way.

ok I have now already replaced the current `on_vnode_stale` with `update_vnode_bitmap`

Reply:
> +1 for passing vnode bitmap to state store.
> 
> > we cannot determine whether a table is singleton from vnode mapping
> 
> why? Can we treat vnode bitmap with all ones as a singleton?

It's not the current vnode bitmap representation for singleton in our system.